tools/daemon/message.c

```diff
@@ -337,3 +357,118 @@ static int proc_msg_state(struct vma_hdr *msg_hdr, size_t size)
 
 	return (sizeof(*data));
 }
+
+static int proc_msg_flow(struct vma_hdr *msg_hdr, size_t size, struct sockaddr_un *peeraddr)
+{
+	int rc = 0;
+	struct vma_msg_flow *data;
+	struct store_pid *pid_value;
+	struct store_flow *value = NULL;
+	struct store_flow *cur_flow = NULL;
+	struct list_head *cur_entry = NULL;
+	int value_new = 0;
+
+	assert(msg_hdr);
+	assert(msg_hdr->code == VMA_MSG_FLOW);
+	assert(size);
+
+	data = (struct vma_msg_flow *)msg_hdr;
+	if (size < sizeof(*data)) {
+		rc = -EBADMSG;
+		goto err;
+	}
+
+	pid_value = hash_get(daemon_cfg.ht, data->hdr.pid);
+	if (NULL == pid_value) {
+		log_error("Failed hash_get() for pid %d errno %d (%s)\n",
+				data->hdr.pid, errno, strerror(errno));
+		rc = -ENOENT;
+		goto err;
+	}
+
+	/* Allocate memory for this value in this place
+	 */
+	value = (void *)calloc(1, sizeof(*value));
+	if (NULL == value) {
+		rc = -ENOMEM;
+		goto err;
+	}
+
+	value->type = data->type;
+	value->if_id = data->if_id;
+	value->tap_id = data->tap_id;
+	switch (value->type) {
+	case VMA_MSG_FLOW_TCP_3T:
+		value->flow.t3.dst_ip = data->flow.t3.dst_ip;
+		value->flow.t3.dst_port = data->flow.t3.dst_port;
+		break;
+	case VMA_MSG_FLOW_TCP_5T:
+		value->flow.t5.src_ip = data->flow.t5.src_ip;
+		value->flow.t5.dst_ip = data->flow.t5.dst_ip;
+		value->flow.t5.src_port = data->flow.t5.src_port;
+		value->flow.t5.dst_port = data->flow.t5.dst_port;
+		break;
+	default:
+		log_error("Received unknown message errno %d (%s)\n", errno,
+				strerror(errno));
+		rc = -EPROTO;
+		goto err;
+	}
```

Comment:
Do we have to copy all that during flow deletion ?

Reply:
not necessary it is just done to unify procedure